src/Ssl/PrivateKey.php

```diff
@@ -37,10 +38,26 @@ public function getResource()
      */
     public function getPublicKey()
     {
-        if (!$resource = openssl_pkey_get_private($this->keyPEM)) {
-            throw new KeyFormatException(sprintf('Fail to convert key into resource: %s', openssl_error_string()));
-        }
+        return new PublicKey(openssl_pkey_get_details($this->getResource())['key']);
+    }
+
+    /**
+     * @param $keyDER
+     *
+     * @return PrivateKey
+     */
+    public static function fromDER($keyDER)
+    {
+        Assert::stringNotEmpty($keyDER, __CLASS__.'::$keyDER should not be an empty string. Got %s');
+
+        $der = base64_encode($keyDER);
+        $lines = str_split($der, 65);
+        $body = implode("\n", $lines);
+        $title = 'PRIVATE KEY';
+        $result = "-----BEGIN {$title}-----\n";
```

Comment:
You can remove the `$title` variable and instead of this concatenation, simply use the implode:

```
$lines = str_split($der, 65);
array_unshift($lines, '-----BEGIN PRIVATE KEY-----');
$lines[] = '-----END PRIVATE KEY-----';
$lines[] = '';

return new self(implode("\n", $lines));
```